Text/Pandoc/Definition.hs

```diff
@@ -209,31 +211,91 @@ instance Eq Format where
 instance Ord Format where
   compare (Format x) (Format y) = compare (T.toCaseFold x) (T.toCaseFold y)
 
+-- | The width of the row head of each row of a table. The row body
+-- takes up the remaining table width.
+type RowHeadColumns = Int
+
+-- | Alignment of a table column.
+data Alignment = AlignLeft
+               | AlignRight
+               | AlignCenter
+               | AlignDefault deriving (Eq, Ord, Show, Read, Typeable, Data, Generic)
+
+-- | The relative width of a column. A @Nothing@ value represents the
+-- default width.
+type ColWidth = Maybe Double
+
+-- | The specification for a single table column.
+type ColSpec = (Alignment, Maybe Double)
+
+-- | A table row. The header cells of the row are determined by the
+-- value of the 'RowHeadColumns' of the table and the ultimate
+-- placement of the table cells on the grid.
+data Row = Row Attr [Cell]
+  deriving (Eq, Ord, Show, Read, Typeable, Data, Generic)
+
+-- | The head of a table.
+type TableHead = [Row]
+
+-- | The body of a table.
+type TableBody = [Row]
+
+-- | The foot of a table.
+type TableFoot = [Row]
+
+-- | The caption of a table, with an optional short caption.
+data Caption = Caption (Maybe [Inline]) [Block]
```

Comment:
are we thinking of potentially reusing this for [figure captions](https://github.com/jgm/pandoc/issues/3177#issuecomment-421639264), or...?

Reply:
I think we should certainly re-use this for figure captions.  It wouldn't make sense for headings.